### PR TITLE
Add Aqua.jl for some static testing

### LINKED ISSUE #1964: Add CI tests to our projects looking for "bad" exports
We sometimes accidentally export non-existing things. E.g. from current Oscar.jl:
```
julia> filter(n->!isdefined(Oscar, n), names(Oscar))
32-element Vector{Symbol}:
 :AbsPrePreSheaf
 :CoherentSheaf
 :DirectProductOfElem
 :LineBundle
 :_non_degeneration_cover
 :affine_patch_type
 :ambient
 :as_smooth_lci_of_cod
 :as_smooth_local_complete_intersection
 :base_scheme_type
 :controlled_transform
 :coset_decomposition
 :covered_patches
 :covering
 :dualsubdivision
 :generator
 :has_complement_class_reps
 :ideal_dict
 :ideal_sheaf_type
 :images_of_variables
 :intersect_stably
 :load_simplicialcomplex
 :normalize
 :p
 :prepare_smooth_center
 :save_simplicialcomplex
 :set_complement_class_reps
 :set_is_finite
 :subdivision_of_vertices
 :total_transform
 :weak_transform
 :weight
```

The `:p` one is due to a typo, @HereAround is already fixing it.

I suggest the following:
- [ ] resolve these (either by adding missing functions or removing the exports)
- [ ] add something like this to our tests:
  ```julia
  @test all(n->isdefined(Oscar, n), names(Oscar))
  ```
- add similar tests to our other packages (AA, Nemo, Hecke, GAP, Singular, Polymake, ...)

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves #1964.

Three of the performed tests (`ambiguities`, `unbound_args`, and `piracy`) currently fail and are thus disabled (with a comment to check if/with how much work they can be fixed).

`deps_compat` requires all dependencies to have a `[compat]` entry, so I added the missing ones. The JuliaRegistry currently does not require those for JLLs (cf. the last paragraph of https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/#Upper-bounded-[compat]-entries), but adding them should not hurt.

`stale_deps` fails on my local setup with some weird error containing the UUID of `GAP_lib_jll`. This can be investigated in a separate PR.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ cohomCalg_jll = "5558cf25-a90e-53b0-b813-cadaa3ae7ade"
 [compat]
 AbstractAlgebra = "0.29.3"
 AlgebraicSolving = "0.3.0"
+Aqua = "0.6"
```
Comment (fingolfin) [inline Project.toml:29]: Do we really need this as a dependency when it is only used by the tests?

Comment (lgoettgens) [reply Project.toml:29]: It is not a dependency but only listed in the `[extra]`. With the compact entry, we can make sure that an `Aqua.jl` release (that probably will change some of the performed tests) does not break our CI.

Comment (lgoettgens) [reply Project.toml:29]: At least in theory. I could unfortunately not find a statement in the docs if `[compact]` is used at all for `[extra]` dependencies.

Comment (fingolfin) [reply Project.toml:29]: yeah, I misphrased that, sorry

Anyway, I think we should move away from the old-style usage of `[extras]` etc in the main `Project.toml` and instead move to a dedicated `test/Project.toml`, that would neatly resolve this issue.

Comment (lgoettgens) [reply Project.toml:29]: I strongly agree. But I think this should be done in a separate PR. I created #2374 to remember.

Comment (thofma) [reply Project.toml:29]: What exactly is the issue? I don't see a problem that is solved by this.

Comment (lgoettgens) [reply Project.toml:29]: > What exactly is the issue? I don't see a problem that is solved by this.

`Aqua.jl` may release a new version `v0.7` in the future that includes some additional test, which fails for OSCAR.
Without this, the second `v0.7` is released, our CI breaks (on all branches). With this change, we can change the used `Aqua.jl` version ourselves, in a controlled environment/PR.

Note:
> At least in theory. I could unfortunately not find a statement in the docs if [compact] is used at all for [extra] dependencies.

Comment (thofma) [reply Project.toml:29]: Yes, the `[compat]` entry is also valid for the `[extra]` entries, which you did here. So everything is fine.

Comment (fingolfin) [reply Project.toml:29]: @lgoettgens agreed (about the separate PR)

**Review by fingolfin — COMMENTED** [review]
Fine by me

**Review by thofma — APPROVED** [review]

**Pull request merged**